_posts/2018-10-21-nabla-k8s.md

```diff
@@ -0,0 +1,288 @@
+```
+
+##  Build and install Nabla runtime `runnc`
+
+```
+Runnc Version Info: v0.2 (commit: 1df1a09c891decaf0bc4279c2322126932863c96)
+```
+
+The first step to running nabla on kubernetes is to run nabla! The first order of business is to make sure we have a running new (v0.2) version of `runnc`. For a step-by-step guide to install `runnc`, take a look at our previous article: [Running a Nabla Container]({% post_url 2018-06-28-nabla-setup %}).
+
+## Setup containerd with `runnc`
+
+```
+Containerd Version Info (commit: 0c5f8f63c3368856c320ae8a1c125e703b73b51d)
+```
+
+We are using [containerd](https://github.com/containerd/containerd)  as a runtime as it supports the selection of an alternative runtime via a pod annotation in kubernetes. This feature is a cri plugin called `untrusted_workload_runtime`.
+
+We enable this feature by modifying a containerd configuration to include the cri plugin:
+
+```
+lumjjb@lumjjb-ThinkPad-P50:~$ cat /etc/containerd/config.toml
+
+subreaper = true
+oom_score = -999
+
+[debug]
+        level = "debug"
+
+[metrics]
+        address = "127.0.0.1:1338"
+
+[plugins.linux]
+        runtime = "runc"
+        shim_debug = true
+
+### ADD THE FOLLOWING CONFIGURATION ###
+[plugins]
+  [plugins.cri.containerd]
+  [plugins.cri.containerd.untrusted_workload_runtime]
+  runtime_type = "io.containerd.runtime.v1.linux"
+  runtime_engine = "/usr/local/bin/runnc"
+```
+
+We then run the containerd service in preparation for setting up kubernetes (Note: This may require running with `root` or `sudo`):
+
+```
+root@lumjjb-ThinkPad-P50:/home/lumjjb# /usr/local/bin/containerd --config /etc/containerd/config.toml
+
+INFO[2018-10-22T21:40:27.299644455-04:00] starting containerd                           revision=de4bb2ddfbb6b2b6a112b3478a935ca74dd7b796 version=v1.2.0-rc.0-44-gde4bb2d
+DEBU[2018-10-22T21:40:27.299684880-04:00] changing OOM score to -999
+INFO[2018-10-22T21:40:27.299892674-04:00] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
+INFO[2018-10-22T21:40:27.299909914-04:00] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
+<<< TRUNCATED >>>
+```
+
+Note that this process runs as a daemon, so we will need to keep this running for
+the rest of the setup. Optionally, you may set it up as a [systemd service](https://manpages.debian.org/stretch/systemd/systemctl.1.en.html).
+
+## Setting up a CNI config
+
+**Note: We are aware of issues with network configurations with `/32` CIDR with ARP-proxying, and are working on upstreaming fixes. [Track Issue](https://github.com/nabla-containers/runnc/issues/40)**
+
+Note: if you already have CNI configured, you may skip this step.
+
+The [Container Network Interface (CNI)](https://github.com/containernetworking/cni) is the interface that kubernetes uses for networking. A configuration is required for our setup using `hack/local-up-cluster.sh`. For simplicity, we use a simple bridge CNI plugin. 
```

Comment:
>I don't really understand what this section is about. Is there a special type of CNI configuration required for nabla? Does it have to be a bridge? Or will any old CNI configuration just work?

It doesn't have to be bridge. Any CNI plugin will do ***

***: Small caveat here is that the conventional CNI plugin creates a veth pair. And there is an unsaid coupling between the CNI and CRI. In the containerd case, the CNI plugin is one of a veth pair. Although there's this small detail, most just see this as a standard CNI plugin since that is what most container runtimes use.